src/AccountingModule.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.28;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeVerifierLib } from "./library/SafeVerifierLib.sol";
+import { AccountingToken } from "./AccountingToken.sol";
+import { IFlexStrategy } from "./FlexStrategy.sol";
+
+interface IAccountingModule {
+    event LowerBoundUpdated(uint16 newValue, uint16 oldValue);
+    event TargetApyUpdated(uint16 newValue, uint16 oldValue);
+
+    error TooEarly();
+    error Unauthorized();
+    error InvariantViolation();
+    error TvlTooLow();
+}
+
+contract AccountingModule is IAccountingModule {
+    using SafeERC20 for IERC20;
+
+    uint256 public constant YEAR = 365.25 days;
+    uint256 public constant DIVISOR = 10_000;
+
+    AccountingToken public immutable ACCOUNTING_TOKEN;
+    address public immutable BASE_ASSET;
+    address public immutable STRATEGY;
+
+    address public SAFE;
+
+    uint16 public cooldownSeconds = 3600;
+    uint64 public nextRewardWindow;
+    uint16 public targetApy; // in bips;
+    uint16 public lowerBound; // in bips; % of tvl
+
+    modifier checkAndResetCooldown() {
+        if (block.timestamp < nextRewardWindow) revert TooEarly();
+        nextRewardWindow = (uint64(block.timestamp) + cooldownSeconds);
+        _;
+    }
+
+    modifier onlyStrategyAllocator() {
+        if (IFlexStrategy(STRATEGY).isAllocator(msg.sender)) revert Unauthorized();
+        _;
+    }
+
+    modifier onlyStrategy() {
+        if (msg.sender != STRATEGY) revert Unauthorized();
+        _;
+    }
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address strategy,
+        address baseAsset,
+        address safe,
+        uint16 targetApy_,
+        uint16 lowerBound_
+    ) {
+        ACCOUNTING_TOKEN = new AccountingToken(name_, symbol_, baseAsset, address(this));
+        BASE_ASSET = baseAsset;
+        STRATEGY = strategy;
+
+        // will revert after having deployed if we proxify this - is there a better way to check?
+        SafeVerifierLib.verify(safe);
+        SAFE = safe;
```

Comment:
Since we're making this ugpradeable, this should be set in intiailize, anything that's not immutable